internal/utils/tls.go

```diff
@@ -0,0 +1,126 @@
+				promConfig.ClientCertPath, promConfig.ClientKeyPath, err)
+		}
+		config.Certificates = []tls.Certificate{cert}
+		logger.Log.Info("Client certificate loaded successfully",
+			"cert_path", promConfig.ClientCertPath, "key_path", promConfig.ClientKeyPath)
+	}
+
+	return config, nil
+}
+
+// ValidateTLSConfig validates TLS configuration.
+// Ensures HTTPS is used and certificate files exist when verification is enabled.
+// Note: This function assumes promConfig is not nil - nil checks should be performed at a higher level.
+func ValidateTLSConfig(promConfig *interfaces.PrometheusConfig) error {
+	// Validate that the URL uses HTTPS (TLS is always required)
+	if !IsHTTPS(promConfig.BaseURL) {
+		return fmt.Errorf("HTTPS is required - URL must use https:// scheme: %s", promConfig.BaseURL)
+	}
+
+	// If InsecureSkipVerify is true, we don't need to validate certificate files
+	// since we're intentionally skipping certificate verification
+	if promConfig.InsecureSkipVerify {
+		logger.Log.Warn("TLS certificate verification is disabled - this is not recommended for production")
+		return nil
+	}
+
+	// Check if certificate files exist (only when not skipping verification)
+	if promConfig.CACertPath != "" {
+		if _, err := os.Stat(promConfig.CACertPath); os.IsNotExist(err) {
+			return fmt.Errorf("CA certificate file not found: %s", promConfig.CACertPath)
+		}
+	}
+
+	if promConfig.ClientCertPath != "" {
+		if _, err := os.Stat(promConfig.ClientCertPath); os.IsNotExist(err) {
+			return fmt.Errorf("client certificate file not found: %s", promConfig.ClientCertPath)
+		}
+	}
+
+	if promConfig.ClientKeyPath != "" {
+		if _, err := os.Stat(promConfig.ClientKeyPath); os.IsNotExist(err) {
+			return fmt.Errorf("client key file not found: %s", promConfig.ClientKeyPath)
+		}
+	}
+
+	return nil
+}
+
+// ParsePrometheusConfigFromEnv parses Prometheus configuration from environment variables.
+// Supports both direct values and file paths for flexible deployment scenarios.
+func ParsePrometheusConfigFromEnv() *interfaces.PrometheusConfig {
+	config := &interfaces.PrometheusConfig{
+		BaseURL: os.Getenv("PROMETHEUS_BASE_URL"),
+	}
+
+	// TLS is always enabled for HTTPS-only support
+	config.InsecureSkipVerify = os.Getenv("PROMETHEUS_TLS_INSECURE_SKIP_VERIFY") == "true"
+	config.CACertPath = os.Getenv("PROMETHEUS_CA_CERT_PATH")
+	config.ClientCertPath = os.Getenv("PROMETHEUS_CLIENT_CERT_PATH")
+	config.ClientKeyPath = os.Getenv("PROMETHEUS_CLIENT_KEY_PATH")
+	config.ServerName = os.Getenv("PROMETHEUS_SERVER_NAME")
+
+	// Support both direct bearer token and token path
+	config.BearerToken = os.Getenv("PROMETHEUS_BEARER_TOKEN")
+	config.TokenPath = os.Getenv("PROMETHEUS_TOKEN_PATH")
+
+	return config
+}
+
+// IsHTTPS checks if the given URL string uses HTTPS scheme
+func IsHTTPS(urlStr string) bool {
+	u, err := url.Parse(urlStr)
+	if err != nil {
+		return false
+	}
+	return u.Scheme == "https"
+}
```

Comment:
Remove this function if this is not reused, and may be you can reduce one unit test. Avoid creating functions that are used just once.

Reply:
removed